### PR TITLE
Stage deploy docs

### PR DESCRIPTION
**Pull request by scholtzan**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: docs/reference/stage-deploys-continuous-integration.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    * Determining dependencies is only relevant for UDFs and views. For queries, available `schema.yaml` files will simply be deployed. 
+    * For UDFs, if a UDF does call another UDF then this UDF needs to be deployed to stage as well.
+    * For views, if a view references another view, table or UDF then each of these referenced artifacts needs to be available on stage as well, otherwise the view cannot even be deployed to stage.
+    * If artifacts are referenced that are not defined as part of the bigquery-etl repo (like stable or live tables) then their schema will get determined and a placeholder `query.sql` file will be created
+    * Also dependencies of dependencies need to be deployed, and so on
+4. Once all artifacts that need to be deployed have been determined, all references to these artifacts in existing SQL files need to be updated. These references will need to point to the stage project and the temporary datasets that artifacts will be published to.
```
Comment (gleonard-m) [inline docs/reference/stage-deploys-continuous-integration.md:20]: Can you clarify how the list of artifacts is determined?

File: docs/cookbooks/common_workflows.md [hunk 2]
```diff
@@ -25,9 +25,9 @@ The [Creating derived datasets tutorial](https://mozilla.github.io/bigquery-etl/
 1. Create a pull request
    * CI fails since table doesn't exist yet
```
Comment (ANich) [inline docs/cookbooks/common_workflows.md:26]: Will CI still fail here?

File: docs/cookbooks/common_workflows.md [hunk 3]
```diff
@@ -103,10 +104,11 @@ Adding a new field to a table schema also means that the field has to propagate
 1. Open a new PR with these changes.
 1. The dry-run-sql task is expected to fail at this point due to mismatch with deployed schemas!
 1. PR reviewed and approved.
-1. Deploy schema changes by running: `./bqetl query schema deploy <dataset>.<table>;`
 1. Find and run again the [CI pipeline](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl?) for the PR.
    * [x] Make sure all dry runs are successful.
 1. Merge pull-request.
+1. Table deploys happen on a nightly cadence through the [`bqetl_artifact_deployment` Airflow DAG](https://workflow.telemetry.mozilla.org/dags/bqetl_artifact_deployment/grid)
```
Comment (ANich) [inline docs/cookbooks/common_workflows.md:110]: See above

**Review by ANich — APPROVED** [review]
Lgtm overall there are a few questions though

File: docs/cookbooks/common_workflows.md (hunk 2 above)
Comment (scholtzan) [reply docs/cookbooks/common_workflows.md:26]: nope

File: docs/reference/stage-deploys-continuous-integration.md (hunk 1 above)
Comment (scholtzan) [reply docs/reference/stage-deploys-continuous-integration.md:20]: I added some more context

**Pull request merged**